src/memphis.ts

```diff
@@ -385,8 +460,12 @@ class Producer {
             headers.headers.set('$memphis_connectionId', this.connection.connectionId);
             headers.headers.set('$memphis_producedBy', this.producerName);
             const subject = this.stationName.replace(/\./g, '#');
-            if (asyncProduce)
-                this.connection.brokerConnection.publish(`${subject}.final`, message, { headers: headers.headers, ackWait: ackWaitSec * 1000 * 1000000 });
+            let schemaData = this.connection.stationSchemaDataMap.get(subject);
+            if (Object.keys(schemaData).length > 0) {
```

Comment:
object keys is a for loop under the hood please try  to access some field instead